cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/DefaultAdmin.java

```diff
@@ -193,12 +193,12 @@ public Void call() throws TopicNotFoundException, IOException {
 
   @Override
   public List<ArtifactInfo> listArtifacts() throws IOException {
-    return defaultArtifactManager.listArtifacts(namespaceId);
+    return defaultArtifactManager.listArtifacts(namespaceId, retryStrategy);
```

Comment:
its kind of weird to pass in the retry strategy here, why not just wrap this using Retries? I think in general, we want the caller to be responsible for retries.